tests/unit/test_charm.py

```diff
@@ -127,14 +131,15 @@ def test_pebble_ready_with_gateway_address_from_config_and_path_routing_mode(sel
         self.assertEqual(self.harness.charm.unit.status, ActiveStatus())
 
         relation = relate(self.harness)
-        for strip_prefix in (False, True):
+        for inject_middleware in (False, True):
             with self.subTest():
                 _requirer_provide_ingress_requirements(
                     harness=self.harness,
                     relation=relation,
                     host="10.1.10.1",
                     port=9000,
-                    strip_prefix=strip_prefix,
+                    strip_prefix=inject_middleware,
+                    redirect_https=inject_middleware,
```

Comment:
shouldn't we test for all combinations?

Reply:
In principle, yes.

For the time being, I wonder. Even adding an additional is pretty extreme, because it suddenly becomes a combinatorics problem with `(n*2)!` (where `n` is the number of possible middlewares) testcases, and mapping it up to anything, even `itertools.product` would probably require `functools.partial`, leaving only the middleware args, and a similar helper to actually "build" the middlewares.

It's a lot of moving pieces which are certainly necessary if/when we add more, but I wonder where it's strictly needed now, since the crucial case of differences (no `stripPrefix` on path-based routing, but `redirectSchema` stays) is actually covered in this patch.